doc/source/reference/random/new-or-different.rst

```diff
@@ -15,7 +15,7 @@ What's New or Different
   streams, use `RandomState`, i.e., `RandomState.gamma` or
   `RandomState.standard_t`.
 
-Quick comparison of legacy `mtrand <legacy>`_ to the new `Generator`
+Quick comparison of legacy :ref:`mtrand <legacy>` to the new `Generator`
```

Comment:
This link wasn't actually broken, but adding the explicit `:ref:` role doesn't hurt.

Reply:
Please see my comment below: https://github.com/numpy/numpy/pull/16472#issuecomment-639450633